extensions/stats/plugin.cc

```diff
@@ -358,6 +358,14 @@ bool PluginRootContext::initializeDimensions(const json& j) {
           if (!name.empty() && name != factory_it.first) {
             continue;
           }
+
+          bool skip_reporting =
+              JsonGetField<bool>(metric, "skip_reporting").value_or(false);
+          if (skip_reporting) {
+            factories.erase(factory_it.first);
```

Comment:
isn't it better to do this in line  293 instead? That is when the factories are created from `defaultMetrics()`

Reply:
That would require moving the parsing of the `metrics` config before L293. But I'm open to better ways (including moving the config around to different spots).

Reply:
Probably not, the current logic also accomodates the case which defines a new metric at higher level (e.g. mesh wide) and skip reporting that at a lower level (e.g. per workload). Also keeping the logic for per metric configuration processing at one place seems cleaner to me.